pkg/models/mappings.go

```diff
@@ -0,0 +1,73 @@
+package models
+
+import (
+	"encoding/json"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+const MappingKind = "TestMocksMapping"
+
+// MocksArray is a custom type that serializes as a comma-separated string
+type MocksArray []string
+
+// MarshalJSON implements json.Marshaler interface
+func (m MocksArray) MarshalJSON() ([]byte, error) {
```

Comment:
Why is MarshalJSON a normal receiver function, whereas UnmarshalJSON is a pointer receiver?  And the same with YAML?